internal/ingest/ingest.go

```diff
@@ -869,15 +871,19 @@ func (ing *Ingester) ingestWorkerLogic(ctx context.Context, provider peer.ID) {
 		ctxIdStr := string(ai.ad.ContextID)
 		// This ad was deleted by a later remove. Push previous onto skips
 		// stack, and set latest skip.
-		if _, ok := rmCtxID[ctxIdStr]; ok {
+		if rmCtxID[ctxIdStr] || rmProviderID[ai.ad.Provider] {
```

Comment:
If the ad is for deleting all data for the provider, then not only should the previous ads in the chain be skipped, but any ads in the entire chain for the provider.  Granted, there should no be any ads for a provider following an ad to delete that provider, but the indexer should catch that case.  If it does not, then it will try to ingest those ads and fail in strange ways.

I would suggest using a separate map of deleted providers, and check that in addition to checking the next ad to skip, [here](https://github.com/filecoin-project/storetheindex/pull/697/files#diff-50d2db719b550ec2183b831d2727d9184635a5647247ea8872ff82d91b6e31bdL904)

Reply:
Makes sense, thanks. Fixed in the latest commit.